DigitalOffice/Tests/CompanyServiceTests/Commands/EditPositionCommandTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using LT.DigitalOffice.CompanyService.Commands;
+using LT.DigitalOffice.CompanyService.Commands.Interfaces;
+using LT.DigitalOffice.CompanyService.Database.Entities;
+using LT.DigitalOffice.CompanyService.Mappers.Interfaces;
+using LT.DigitalOffice.CompanyService.Repositories.Interfaces;
+using LT.DigitalOffice.CompanyService.RestRequests;
+using Moq;
+using NUnit.Framework;
+using System;
+using FluentValidation;
+
+namespace LT.DigitalOffice.CompanyServiceUnitTests.Commands
+{
+    class EditPositionCommandTests
+    {
+        private IEditPositionCommand command;
+        private Mock<ICompanyRepository> repositoryMock;
+        private Mock<IMapper<EditPositionRequest, DbPosition>> mapperMock;
+
+        private DbPosition newPosition;
+        private EditPositionRequest request;
+
+        private string Name = "Position";
+        private string Description = "Description";
+
+        private Mock<IValidator<EditPositionRequest>> validatorMock;
+
+        [OneTimeSetUp]
+        public void Initialization()
+        {
+            validatorMock = new Mock<IValidator<EditPositionRequest>>();
+            repositoryMock = new Mock<ICompanyRepository>();
+            mapperMock = new Mock<IMapper<EditPositionRequest, DbPosition>>();
+
+            command = new EditPositionCommand(validatorMock.Object, repositoryMock.Object, mapperMock.Object);
+
+            request = new EditPositionRequest
+            {
+                Id = Guid.NewGuid(),
+                Name = Name,
+                Description = Description
+            };
+
+            newPosition = new DbPosition
+            {
+                Id = request.Id,
+                Name = request.Name,
+                Description = request.Description
+            };
+        }
+
+        [Test]
+        public void FailEditPositionIncorrectPositionData()
+        {
+            validatorMock
+                .Setup(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
+
+            Assert.Throws<ValidationException>(() => command.Execute(request), "Position field validation error");
```

Comment:
It would be nice to make sure that the repository was not called.

Reply:
Fixed